src/services/users.js

```diff
@@ -332,6 +360,16 @@ module.exports = class UserHelper {
 			//If role is changed, the role change, org policy changes for that user
 			//and additional data update of the user is done by orgAdmin's roleChange workflow
 			const roleChangeResult = await orgAdminService.roleChange(roleChangePayload, userExtensionData, tenantCode)
+
+			// Invalidate cache after role change
+			try {
+				await cacheHelper.mentee.delete(tenantCode, userExtensionData.organization.code, userExtensionData.id)
```

Comment:
@sumanvpacewisdom if the first statement which mentee cache failed delete then, it won't try to delete it from the mentor cache. 
Change the logic, use try catch block for each cache deletion. Check in other places do the same.